src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/WatchImage.cs

```diff
@@ -49,15 +49,26 @@ private void NodeModelOnPropertyChanged(object sender, PropertyChangedEventArgs
                 return;
 
             var bitmap = data.Data as Bitmap;
-            
-            nodeView.Dispatcher.BeginInvoke(new Action<Bitmap>(SetImageSource), new object[] { bitmap });
+            if (bitmap != null)
+            {
+                nodeView.Dispatcher.BeginInvoke(new Action<Bitmap>(SetImageSource), new object[] { bitmap });
+            }
         }
 
         private void SetImageSource(Bitmap bmp)
         {
-            using (bmp)
+            try
+            {
+                if (bmp != null)
+                    image.Source = ResourceUtilities.ConvertToImageSource(bmp);
+            }
+            catch (Exception e)
+            {
+            }
+            finally 
             {
-                image.Source = ResourceUtilities.ConvertToImageSource(bmp);
+                bmp.Dispose();
+                bmp = null;
```

Comment:
Two things... if `bmp` is `null`, doesn't it crash `bmp.Dispose()` call in `finally`? Also, why don't we do `if (bmp == null)` prior to the `using` instead of changing these into `try-catch` block?